qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1856,6 +1867,16 @@ def __json_encode__(self):
 
         # Handle non-serializable objects
         json_value["_jobs"] = self._safe_serialize_jobs()
+        json_value["_data"] = self._safe_serialize_data()
```

Comment:
From my understanding of the `json_equiv` function this shouldn't be necessary and it should still match as equiv for data containing dict or `Count` objects even if `Count` objects get round-tripped as keys

Reply:
This line was removed from the code. Resolving conversation.